bridges/snowbridge/primitives/core/src/outbound/v2.rs

```diff
@@ -0,0 +1,348 @@
+		token_id: H256,
+		/// Name of the token
+		name: Vec<u8>,
+		/// Short symbol for the token
+		symbol: Vec<u8>,
+		/// Number of decimal places
+		decimals: u8,
+	},
+	/// Mint foreign token from Polkadot
+	MintForeignToken {
+		/// ID for the token
+		token_id: H256,
+		/// The recipient of the newly minted tokens
+		recipient: H160,
+		/// The amount of tokens to mint
+		amount: u128,
+	},
+}
+
+impl Command {
+	/// Compute the enum variant index
+	pub fn index(&self) -> u8 {
+		match self {
+			Command::Upgrade { .. } => 0,
+			Command::SetOperatingMode { .. } => 1,
+			Command::UnlockNativeToken { .. } => 2,
+			Command::RegisterForeignToken { .. } => 3,
+			Command::MintForeignToken { .. } => 4,
+			Command::CreateAgent { .. } => 5,
+		}
+	}
+
+	/// ABI-encode the Command.
+	pub fn abi_encode(&self) -> Vec<u8> {
+		match self {
+			Command::Upgrade { impl_address, impl_code_hash, initializer, .. } => UpgradeParams {
+				implAddress: Address::from(impl_address.as_fixed_bytes()),
+				implCodeHash: FixedBytes::from(impl_code_hash.as_fixed_bytes()),
+				initParams: initializer.clone().map_or(vec![], |i| i.params),
+			}
+			.abi_encode(),
+			Command::CreateAgent { agent_id } =>
+				CreateAgentParams { agentID: FixedBytes::from(agent_id.as_fixed_bytes()) }
+					.abi_encode(),
+			Command::SetOperatingMode { mode } =>
+				SetOperatingModeParams { mode: (*mode) as u8 }.abi_encode(),
+			Command::UnlockNativeToken { token, recipient, amount, .. } =>
+				UnlockNativeTokenParams {
+					token: Address::from(token.as_fixed_bytes()),
+					recipient: Address::from(recipient.as_fixed_bytes()),
+					amount: *amount,
+				}
+				.abi_encode(),
+			Command::RegisterForeignToken { token_id, name, symbol, decimals } =>
+				RegisterForeignTokenParams {
+					foreignTokenID: FixedBytes::from(token_id.as_fixed_bytes()),
+					name: name.to_vec(),
+					symbol: symbol.to_vec(),
+					decimals: *decimals,
+				}
+				.abi_encode(),
+			Command::MintForeignToken { token_id, recipient, amount } => MintForeignTokenParams {
+				foreignTokenID: FixedBytes::from(token_id.as_fixed_bytes()),
+				recipient: Address::from(recipient.as_fixed_bytes()),
+				amount: *amount,
+			}
+			.abi_encode(),
+		}
+	}
+}
+
+/// Representation of a call to the initializer of an implementation contract.
+/// The initializer has the following ABI signature: `initialize(bytes)`.
+#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct Initializer {
+	/// ABI-encoded params of type `bytes` to pass to the initializer
+	pub params: Vec<u8>,
+	/// The initializer is allowed to consume this much gas at most.
+	pub maximum_required_gas: u64,
+}
+
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(PartialEq))]
+/// Fee for delivering message
+pub struct Fee<Balance>
+where
+	Balance: BaseArithmetic + Unsigned + Copy,
+{
+	/// Fee to cover cost of processing the message locally
+	pub local: Balance,
+}
```

Comment:
This type seems unnecessary for V2?

Reply:
https://github.com/yrong/polkadot-sdk/pull/4/commits/fb3b30c348dc7c31022324313139ae18b09ef237